packages/playwright-core/src/server/page.ts

```diff
@@ -631,3 +642,57 @@ function addPageBinding(bindingName: string, needsHandle: boolean) {
   };
   (globalThis as any)[bindingName].__installed = true;
 }
+
+class FrameThrottler {
+  private _acks: (() => void)[] = [];
+  private _interval: number;
+  private _nonThrottledFrames: number;
+  private _budget: number;
+  private _intervalId: NodeJS.Timeout | undefined;
+
+  constructor(nonThrottledFrames: number, interval: number) {
+    this._nonThrottledFrames = nonThrottledFrames;
+    this._budget = nonThrottledFrames;
+    this._interval = interval;
+  }
+
+  setEnabled(enabled: boolean) {
+    if (enabled) {
+      if (this._intervalId)
+        clearInterval(this._intervalId);
+      this._intervalId = setInterval(() => this._tick(), this._interval);
+    } else if (this._intervalId) {
+      clearInterval(this._intervalId);
+      this._intervalId = undefined;
+    }
+  }
+
+  recharge() {
+    // Send all acks, reset budget.
+    for (const ack of this._acks)
+      ack();
+    this._acks = [];
+    this._budget = this._nonThrottledFrames;
+  }
+
+  ack(ack: () => void) {
+    // Either not engaged or video is also recording, don't throttle.
```

Comment:
I didn't get where we disable this when video is recording.

Reply:
We don't!